upgrade/upgrades: TestCanaryStatsDelayDelete failed

upgrade/upgrades.TestCanaryStatsDelayDelete [failed](https://mesolite.cluster.engflow.com/invocations/default/80d7bbf2-bfa1-41a4-8f96-a25d095e8c05?testReportRun=1&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvdXBncmFkZS91cGdyYWRlczp1cGdyYWRlc190ZXN0) on master @ [f19780c8f0a41a77a10751cfa2e987da03be28a8](https://github.com/cockroachdb/cockroach/commits/f19780c8f0a41a77a10751cfa2e987da03be28a8):


```
=== RUN   TestCanaryStatsDelayDelete
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestCanaryStatsDelayDelete489985163
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    test_server_shim.go:150: cluster virtualization disabled due to issue: #156332 (expected label: C-bug)
    v26_2_add_table_statistics_delay_delete_column_test.go:223: 
        	Error Trace:	pkg/upgrade/upgrades/v26_2_add_table_statistics_delay_delete_column_test.go:223
        	Error:      	"0" is not greater than "0"
        	Test:       	TestCanaryStatsDelayDelete
        	Messages:   	after upgrade: expected some stats with delay_delete=true when canary_window is set
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestCanaryStatsDelayDelete489985163
--- FAIL: TestCanaryStatsDelayDelete (68.39s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/release-eng @cockroachdb/release-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCanaryStatsDelayDelete.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62760

upgrade/upgrades.TestCanaryStatsDelayDelete [failed](https://mesolite.cluster.engflow.com/invocations/default/90d8b4d7-4ba4-4048-b89f-2d7931169190?testReportRun=1&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvdXBncmFkZS91cGdyYWRlczp1cGdyYWRlc190ZXN0) on master @ [a8d03b328a02b24f7ed20c9447b98a1b34bcce9c](https://github.com/cockroachdb/cockroach/commits/a8d03b328a02b24f7ed20c9447b98a1b34bcce9c):


```
=== RUN   TestCanaryStatsDelayDelete
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestCanaryStatsDelayDelete1914722061
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    test_server_shim.go:150: cluster virtualization disabled due to issue: #156332 (expected label: C-bug)
    v26_2_add_table_statistics_delay_delete_column_test.go:223: 
        	Error Trace:	pkg/upgrade/upgrades/v26_2_add_table_statistics_delay_delete_column_test.go:223
        	Error:      	"0" is not greater than "0"
        	Test:       	TestCanaryStatsDelayDelete
        	Messages:   	after upgrade: expected some stats with delay_delete=true when canary_window is set
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestCanaryStatsDelayDelete1914722061
--- FAIL: TestCanaryStatsDelayDelete (65.36s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCanaryStatsDelayDelete.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/upgrade/upgrades/v26_2_add_table_statistics_delay_delete_column_test.go
+++ b/pkg/upgrade/upgrades/v26_2_add_table_statistics_delay_delete_column_test.go
@@ -132,9 +132,17 @@ func TestCanaryStatsDelayDelete(t *testing.T) {
 	)
 	defer tc.Stopper().Stop(ctx)
 
+	// Disable automatic stats collection to prevent it from racing with
+	// manual ANALYZE calls. The auto refresher would create __auto__ stats
+	// and delete the test's manually-created stats, breaking assertions.
+	_, err := sqlDB.Exec(
+		`SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false`,
+	)
+	require.NoError(t, err)
+
 	// Verify that sql_stats_canary_window is rejected before upgrading
 	// to V26_2.
-	_, err := sqlDB.Exec(
+	_, err = sqlDB.Exec(
 		`CREATE TABLE t_canary (k INT PRIMARY KEY) WITH (sql_stats_canary_window = '15s')`,
 	)
 	require.Error(t, err)

